src/iterable/flattenDeep.js

```diff
@@ -0,0 +1,29 @@
+"use strict";
+
+/**
+ * Yields a flatten array where the deep level is specified.
+ * flattenDeep(1) will have the same behavior as flatten().
+ *
+ * @this {Iterable<T>}
```

Comment:
The `this` type of this is just `Iterable`, the return type depends on the depth and thus can't be expressed nicely with flow's type system.